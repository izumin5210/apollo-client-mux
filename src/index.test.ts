import { ApolloCache, ApolloClient, InMemoryCache } from "@apollo/client/core";
import { SchemaLink } from "@apollo/client/link/schema";
import { beforeEach, expect, test } from "vitest";

import { createApolloLinkMux, ApolloCacheMux } from ".";
import {
  fragment1,
  fragment2,
  query1,
  query2,
  schema1,
  schema2,
} from "./test/fixtures";

// biome-ignore lint/suspicious/noExplicitAny: <explanation>
let cache: ApolloCache<any>;
// biome-ignore lint/suspicious/noExplicitAny: <explanation>
let client: ApolloClient<any>;

beforeEach(() => {
  cache = new ApolloCacheMux({
    defaultCache: new InMemoryCache(),
    caches: {
      graph2: new InMemoryCache(),
    },
  });

  client = new ApolloClient({
    cache,
    ssrMode: true,
    link: createApolloLinkMux({
      links: {
        graph2: new SchemaLink({ schema: schema2 }),
      },
      defaultLink: new SchemaLink({ schema: schema1 }),
    }),
  });
});

test("the request is branched by the @endpoint directive, and the cache key is separated", async () => {
  expect(await client.query({ query: query1 })).toMatchSnapshot();
  expect(await client.query({ query: query2 })).toMatchSnapshot();

  expect(
    client.readFragment({
      fragment: fragment1,
      id: "User:1",
    }),
  ).toMatchSnapshot();

  expect(
    client.readFragment({
      fragment: fragment2,
      id: "User:1",
    }),
  ).toMatchSnapshot();

  expect(cache.readQuery({ query: query1 })).toMatchSnapshot();
  expect(cache.readQuery({ query: query2 })).toMatchSnapshot();
  expect(cache.extract()).toMatchSnapshot();
});
