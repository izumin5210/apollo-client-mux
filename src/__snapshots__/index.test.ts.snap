// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`the request is branched by the @endpoint directive, and the cache key is separated 1`] = `
{
  "data": {
    "users": [
      {
        "__typename": "User",
        "email": "alice@example.com",
        "id": "1",
        "name": "Alice",
      },
      {
        "__typename": "User",
        "email": "bob@example.com",
        "id": "2",
        "name": "Bob",
      },
      {
        "__typename": "User",
        "email": "charlie@example.com",
        "id": "3",
        "name": "Charlie",
      },
    ],
  },
  "loading": false,
  "networkStatus": 7,
}
`;

exports[`the request is branched by the @endpoint directive, and the cache key is separated 2`] = `
{
  "data": {
    "users": [
      {
        "__typename": "User",
        "id": "1",
        "login": "alice",
        "name": "Alice",
      },
      {
        "__typename": "User",
        "id": "2",
        "login": "bob",
        "name": "Bob",
      },
      {
        "__typename": "User",
        "id": "3",
        "login": "charlie",
        "name": "Charlie",
      },
    ],
  },
  "loading": false,
  "networkStatus": 7,
}
`;

exports[`the request is branched by the @endpoint directive, and the cache key is separated 3`] = `
{
  "__typename": "User",
  "email": "alice@example.com",
  "id": "1",
  "name": "Alice",
}
`;

exports[`the request is branched by the @endpoint directive, and the cache key is separated 4`] = `
{
  "__typename": "User",
  "id": "1",
  "login": "alice",
  "name": "Alice",
}
`;

exports[`the request is branched by the @endpoint directive, and the cache key is separated 5`] = `
{
  "users": [
    {
      "__typename": "User",
      "email": "alice@example.com",
      "id": "1",
      "name": "Alice",
    },
    {
      "__typename": "User",
      "email": "bob@example.com",
      "id": "2",
      "name": "Bob",
    },
    {
      "__typename": "User",
      "email": "charlie@example.com",
      "id": "3",
      "name": "Charlie",
    },
  ],
}
`;

exports[`the request is branched by the @endpoint directive, and the cache key is separated 6`] = `
{
  "users": [
    {
      "__typename": "User",
      "id": "1",
      "login": "alice",
      "name": "Alice",
    },
    {
      "__typename": "User",
      "id": "2",
      "login": "bob",
      "name": "Bob",
    },
    {
      "__typename": "User",
      "id": "3",
      "login": "charlie",
      "name": "Charlie",
    },
  ],
}
`;

exports[`the request is branched by the @endpoint directive, and the cache key is separated 7`] = `
{
  "default": {
    "ROOT_QUERY": {
      "__typename": "Query",
      "users": [
        {
          "__ref": "User:1",
        },
        {
          "__ref": "User:2",
        },
        {
          "__ref": "User:3",
        },
      ],
    },
    "User:1": {
      "__typename": "User",
      "email": "alice@example.com",
      "id": "1",
      "name": "Alice",
    },
    "User:2": {
      "__typename": "User",
      "email": "bob@example.com",
      "id": "2",
      "name": "Bob",
    },
    "User:3": {
      "__typename": "User",
      "email": "charlie@example.com",
      "id": "3",
      "name": "Charlie",
    },
  },
  "namespaced": {
    "graph2": {
      "ROOT_QUERY": {
        "__typename": "Query",
        "users": [
          {
            "__ref": "User:1",
          },
          {
            "__ref": "User:2",
          },
          {
            "__ref": "User:3",
          },
        ],
      },
      "User:1": {
        "__typename": "User",
        "id": "1",
        "login": "alice",
        "name": "Alice",
      },
      "User:2": {
        "__typename": "User",
        "id": "2",
        "login": "bob",
        "name": "Bob",
      },
      "User:3": {
        "__typename": "User",
        "id": "3",
        "login": "charlie",
        "name": "Charlie",
      },
    },
  },
}
`;
